tidy3d/components/data/monitor_data.py

```diff
@@ -996,6 +1009,111 @@ def to_source(
             field_dataset=dataset, source_time=source_time, center=center, size=size, **kwargs
         )
 
+    def make_adjoint_sources(
+        self, dataset_names: list[str]
+    ) -> List[Union[CustomCurrentSource, PointDipole]]:
+        """Converts a :class:`.FieldData` to a list of adjoint current or point sources."""
+
+        if np.allclose(np.array(self.monitor.size), np.zeros(3)):
+            return self.to_adjoint_point_sources()
+
+        return self.to_adjoint_field_sources()
+
+    def to_adjoint_point_sources(self) -> List[PointDipole]:
+        """Create adjoint point dipole source if this field data contains one item."""
+
+        sources = []
+
+        for polarization, field_component in self.field_components.items():
+            if field_component is None:
+                continue
+
+            for freq0 in field_component.coords["f"]:
+                omega0 = 2 * np.pi * freq0
+                scaling_factor = 1 / (MU_0 * omega0)
+
+                forward_amp = complex(np.squeeze(field_component.sel(f=freq0).values))
```

Comment:
Seems like `complex()` casting here wouldn't be necessary? It goes directly into `np.angle`, i.e. if these values were real before the cast the phase would still be zero, and `np.angle` accepts real arguments.

Reply:
I think it was partially to cast it out of a `np.ndarray`. 

How about I use `self.get_amplitude(field_component.sel(f=freq0))`, which is making use of the helper function I have in the other 2 adjoint source methods for consistency?